### PR TITLE
fix(s3api/sts): respect MaxSessionLength config in DurationSeconds validation

### LINKED ISSUE #11265: AssumeRoleWithWebIdentity DurationSeconds hardcoded to 43200s max, ignores maxSessionLength from iam.json
## Title
AssumeRoleWithWebIdentity DurationSeconds hardcoded to 43200s max, ignores maxSessionLength from iam.json

## Environment
- SeaweedFS 79b872021 / 4.45, `weed server -s3 -s3.iam.config=/etc/sw/iam.json`
- OIDC provider: Keycloak

## Problem
`weed/s3api/s3api_sts.go` hardcodes:

```go
maxDurationSeconds = int64(43200)  // 12 hours (AssumeRole)
```

`parseDurationSeconds()` validates against this constant, not against `sts.maxSessionLength` from `iam.json`. So setting `maxSessionLength` above 12h has no effect on `AssumeRoleWithWebIdentity` / `AssumeRole` / `AssumeRoleWithLDAPIdentity`.

For comparison, MinIO's `AssumeRoleWithWebIdentity` supports `DurationSeconds` up to 7 days (some versions up to 365 days), configurable — not a fixed 12h ceiling.

I understand that this is AWS constants, but is it necessary to be bound to AWS constraints here ?

## Repro
`iam.json`: `"maxSessionLength": "168h"`

```bash
aws sts assume-role-with-web-identity --role-arn arn:aws:iam::role/S3UserRole \
  --role-session-name test --web-identity-token "$TOKEN" \
  --duration-seconds 604800 --endpoint-url https://s3.example.com
```

**Expected:** session up to `maxSessionLength`.

**Actual:**

STS error response: code=InvalidParameterValue, type=Sender, message=DurationSeconds must be between 900 and 43200 seconds

Confirmed server-side (server log `s3api_sts.go:1153`), not client SDK validation.

## Suggested fix
Derive the bound from `h.stsService.Config.MaxSessionLength` instead of the hardcoded constant (fallback to 43200s default when unset) — same idea already used for `GetFederationToken`'s separate 36h bound.

## Related
#11039 / #11048 touch the same file/area recently.

### PR DESCRIPTION
**Pull request by chrislusf**
## Summary

Fixes #11265.

`AssumeRole`, `AssumeRoleWithWebIdentity`, and `AssumeRoleWithLDAPIdentity` validated `DurationSeconds` against a hardcoded `43200`s (12h) ceiling in **two** places, so raising `maxSessionLength` in `iam.json` above 12h had no effect — requests were rejected with:

```
DurationSeconds must be between 900 and 43200 seconds
```

The two gates:
1. **HTTP handler** — `parseDurationSeconds` (`weed/s3api/s3api_sts.go`) rejected before the request reached the service.
2. **STS service** — `validateSessionDurationSeconds` (`weed/iam/sts/sts_service.go`) rejected again inside `AssumeRoleWithWebIdentity` / `AssumeRoleForPrincipal` / `AssumeRoleWithCredentials` (both the IAMManager-dispatched and bare-STS paths go through here).

This PR derives the upper bound from the configured STS `MaxSessionLength` in both gates, falling back to `DefaultMaxSessionLength` (`43200`s) when unset. When `MaxSessionLength` is configured below the 900s API minimum, the default bound is kept so the previous capping behavior (via `calculateSessionDuration`) is preserved instead of creating an empty valid range. The service layer (`calculateSessionDuration`) already caps the issued duration at `MaxSessionLength`, so this only relaxes the input-validation gates.

### Commits (one per logic change)
- `Refactor parseDurationSeconds into a STSHandlers method` — pure refactor, no behavior change; lets the handler validator reach the STS config.
- `Respect MaxSessionLength config in STS DurationSeconds validation` — handler-layer fix.
- `Add tests for STS DurationSeconds MaxSessionLength bound` — handler-layer tests.
- `Refactor validateSessionDurationSeconds into a STSService method` — pure refactor, no behavior change; lets the service validator reach the STS config.
- `Respect MaxSessionLength config in STS service DurationSeconds validation` — service-layer fix (the second gate flagged by review).
- `Add tests for STS service DurationSeconds MaxSessionLength bound` — service-layer tests.
- `Preserve capping when MaxSessionLength is below the API minimum` — guard against an empty valid range for sub-900s configs (review feedback).
- `Add tests for sub-minimum MaxSessionLength capping behavior` — tests for the guard.

#### Test plan
- [x] `go build ./weed/iam/sts/... ./weed/s3api/...`
- [x] `go test ./weed/iam/sts/... ./weed/s3api/...`
- [ ] Repro from the issue: with `iam.json` `maxSessionLength: "168h"`, `aws sts assume-role-with-web-identity --duration-seconds 604800` succeeds (previously rejected).

## Summary by CodeRabbit

- **New Features**
  - STS session duration limits now respect the configured maximum session length.
  - When no valid custom maximum is configured, the service uses its default maximum.
  - The minimum session duration remains 900 seconds, including when a configured maximum is lower than this threshold.

- **Bug Fixes**
  - Improved consistency in duration validation across supported STS authentication flows.

### CODE REVIEW

File: weed/s3api/s3api_sts.go [hunk 1]
```diff
@@ -64,9 +63,14 @@ func parseDurationSeconds(r *http.Request) (*int64, STSErrorCode, error) {
 		return nil, STSErrInvalidParameterValue, fmt.Errorf("invalid DurationSeconds: %w", err)
 	}
 
-	if ds < minDurationSeconds || ds > maxDurationSeconds {
+	maxDuration := int64(sts.DefaultMaxSessionLength)
+	if h.stsService != nil && h.stsService.Config != nil && h.stsService.Config.MaxSessionLength.Duration > 0 {
+		maxDuration = int64(h.stsService.Config.MaxSessionLength.Duration / time.Second)
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/s3api/s3api_sts.go]: Addressed. The service-layer `validateSessionDurationSeconds` now also derives its upper bound from the configured `MaxSessionLength` (fallback to `DefaultMaxSessionLength`), so the handler and service gates agree. See commits `9cee7106f` (refactor to a STSService method) and `b56ef8bf0` (use MaxSessionLength).

File: weed/s3api/s3api_sts.go [hunk 2]
```diff
@@ -131,9 +131,13 @@ func parseDurationSecondsWithBounds(r *http.Request, minSec, maxSec int64) (*int
...
-// parseDurationSeconds parses DurationSeconds for AssumeRole (15 min to 12 hours)
-func parseDurationSeconds(r *http.Request) (*int64, STSErrorCode, error) {
-	return parseDurationSecondsWithBounds(r, minDurationSeconds, maxDurationSeconds)
+// parseDurationSeconds parses DurationSeconds for AssumeRole (15 min to MaxSessionLength)
+func (h *STSHandlers) parseDurationSeconds(r *http.Request) (*int64, STSErrorCode, error) {
+	maxSec := maxDurationSeconds
+	if h.stsService != nil && h.stsService.Config != nil && h.stsService.Config.MaxSessionLength.Duration > 0 {
+		maxSec = int64(h.stsService.Config.MaxSessionLength.Duration / time.Second)
+	}
+	return parseDurationSecondsWithBounds(r, minDurationSeconds, maxSec)
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/s3api/s3api_sts.go:143]: Addressed. The service-layer `validateSessionDurationSeconds` now also derives its upper bound from the configured `MaxSessionLength` (fallback to `DefaultMaxSessionLength`), so the handler and service gates agree. See commits `9cee7106f` (refactor to a STSService method) and `b56ef8bf0` (use MaxSessionLength).

[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/s3api/s3api_sts.go:143]: Addressed. The service-layer `validateSessionDurationSeconds` now also derives its upper bound from the configured `MaxSessionLength` (fallback to `DefaultMaxSessionLength`), so the handler and service gates agree. See commits `9cee7106f` (refactor to a STSService method) and `b56ef8bf0` (use MaxSessionLength).

File: weed/s3api/s3api_sts.go [hunk 3]
```diff
@@ -64,9 +63,14 @@ func parseDurationSeconds(r *http.Request) (*int64, STSErrorCode, error) {
 		return nil, STSErrInvalidParameterValue, fmt.Errorf("invalid DurationSeconds: %w", err)
 	}
 
-	if ds < minDurationSeconds || ds > maxDurationSeconds {
+	maxDuration := int64(sts.DefaultMaxSessionLength)
+	if h.stsService != nil && h.stsService.Config != nil && h.stsService.Config.MaxSessionLength.Duration > 0 {
+		maxDuration = int64(h.stsService.Config.MaxSessionLength.Duration / time.Second)
+	}
+
+	if ds < minDurationSeconds || ds > maxDuration {
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/s3api/s3api_sts.go]: Good catch — addressed in commit `069aa739b`. When `MaxSessionLength` is configured below the 900s API minimum, the validators now keep the default upper bound instead of creating an empty valid range, so explicit `DurationSeconds` within the default range are still accepted and `calculateSessionDuration` caps the issued duration to the shorter configured limit (preserving the previous capping behavior). Added tests in `8fd8b15d2`.
